### PR TITLE
[MIBM-465] Add optional 'take proof' message

### PR DESCRIPTION
**Pull request by andy1138**

### CODE REVIEW

File: test/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationControllerSpec.scala [hunk 1]
```diff
@@ -79,4 +80,65 @@ class DeclarationConfirmationControllerSpec extends DeclarationJourneyController
     status(eventualResult) mustBe 303
     redirectLocation(eventualResult) mustBe Some("/declare-commercial-goods/cannot-access-page")
   }
+
+  private def generateDeclarationConfirmationPage(decType: DeclarationType, purchaseAmount: Long): Future[Result] = {
```
[the remark this answers is not in the document]
Comment (sureshhmrc) [reply test/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationControllerSpec.scala:84]: Also the specs seem to be failing

File: conf/messages.cy [hunk 2]
```diff
@@ -432,8 +432,9 @@ declarationConfirmation.ul.1 = mynd drwy’r
 declarationConfirmation.ul.1.strong = sianel wyrdd
 declarationConfirmation.ul.1.1 = (dim byd i’w ddatgan) wrth y tollau
 declarationConfirmation.ul.2 = mynd â’r cadarnhad o ddatganiad hwn gyda chi
-declarationConfirmation.Import.ul.3 = mynd â’r anfonebau neu dderbynebau ar gyfer yr holl nwyddau rydych yn dod â nhw i mewn i Brydain Fawr
-declarationConfirmation.Export.ul.3 = mynd â’r anfonebau neu’r derbynebau ar gyfer yr holl nwyddau rydych yn mynd â nhw allan o Brydain Fawr
+declarationConfirmation.ul.3 = take proof which shows where the EU goods were produced
```
[the remark this answers is not in the document]
Comment (andy1138) [reply conf/messages.cy:435]: Only the key has changed from `declarationConfirmation.Export.ul.3` to `declarationConfirmation.Export.ul.4` to allow for the new line of text to be added to the list. We could add the new line without a index number.

File: test/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationControllerSpec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (andy1138) [reply test/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationControllerSpec.scala:84]: Moving to CoreTestData will be tricky as it uses methods in the base classes, but will have a look.

Don't know why the test fail on jenkins as it's working local, Its coming back with status 303.

[the remark this answers is not in the document]
Comment (andy1138) [reply test/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationControllerSpec.scala]: Move to CoreTestData and now works on Jenkins

File: test/uk/gov/hmrc/merchandiseinbaggage/CoreTestData.scala [hunk 3]
```diff
@@ -193,4 +197,51 @@ trait CoreTestData {
...
+
+    val journey: DeclarationJourney = completedDeclarationJourney
+      .copy(sessionId = sessionId, declarationType = decType, createdAt = created, declarationId = id)
+
+    val persistedDeclaration = journey.declarationIfRequiredAndComplete.map { declaration =>
+      if (decType == DeclarationType.Import) declaration.copy(maybeTotalCalculationResult = Some(totalCalculationResult))
+      else declaration
+    }
+
+    val declarationConfirmationView = new DeclarationConfirmationView(layout, null, link)
```
Comment (sureshhmrc) [inline test/uk/gov/hmrc/merchandiseinbaggage/CoreTestData.scala:241]: i think you can get this View from Injector directly without having to create an instance like this ?

**Review by sureshhmrc — APPROVED** [review]

**Pull request merged**